### PR TITLE
Incorrectly baked model crashes Interface

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
https://highfidelity.manuscript.com/f/cases/14329/
An FBX model (Mason_Idle.fbx - attached to ticket) works correctly when loaded directly via the create-entity menu, but causes the Interface to crash when uploaded to the asset server, baked, and added to world from the assets menu.

The immediate cause of the crash is the baked fbx file being seemingly half-empty and FBXReader::extractFBXGeometry() making certain assumptions about objects and their links. The fbx file appears half-empty because it contains extra null nodes causing our reader to skip sections.

This PR has:

1. Simple fix for interface crash if there are no joints and there should be.

2. Fix for skipping nodes when a spurious null node is present - FBXReader ignores null nodes rather than terminating the set of siblings. The extent of the sibling  set is known so we don't need to rely on the terminating null.

3. Don't generate the null node by not changing the parent node if the mesh manipulation fails. For some reason this was the behaviour if the Draco operation produced errors or warnings, but not if it silently failed. I think this is an error.

4. I added a class to convert our internal FBXNode representation to a more-readable JSON form. Also optional code to spit out JSON forms of baked and unbaked fbx files.

5. Tweaked the fbx output header so blender accepts it.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9597/HighFidelity-Beta-PR13127-82e43aea113116ca917294d9cda360e16d5456ab.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9597/HighFidelity-Beta-Interface-PR13127-82e43aea113116ca917294d9cda360e16d5456ab.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9608/app_13127_PR_df6609c1f56bfb45fcc889bc0ecf082865e3590a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9608/HighFidelity-Beta-PR13127-df6609c1f56bfb45fcc889bc0ecf082865e3590a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9608/HighFidelity-Beta-Interface-PR13127-df6609c1f56bfb45fcc889bc0ecf082865e3590a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9608/HighFidelity-Beta-PR13127-df6609c1f56bfb45fcc889bc0ecf082865e3590a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9608/HighFidelity-Beta-Interface-PR13127-df6609c1f56bfb45fcc889bc0ecf082865e3590a.exe

File: libraries/fbx/src/FBXWriter.cpp [hunk 1]
```diff
@@ -62,8 +62,7 @@ QByteArray FBXWriter::encodeFBX(const FBXNode& root) {
     out.setVersion(QDataStream::Qt_4_5);
 
     out.writeRawData(FBX_BINARY_PROLOG, FBX_BINARY_PROLOG.size());
-    auto bytes = QByteArray(FBX_HEADER_BYTES_BEFORE_VERSION - FBX_BINARY_PROLOG.size(), '\0');
-    out.writeRawData(bytes, bytes.size());
+    out.writeRawData("\0\x1a", 3);  // Blender needs this header component.
```
Comment (hyperlogic) [inline libraries/fbx/src/FBXWriter.cpp]: Magic number, please make this a constant.

File: libraries/baking/src/FBXBaker.cpp [hunk 2]
```diff
@@ -345,3 +365,43 @@ void FBXBaker::rewriteAndBakeSceneTextures() {
         }
     }
 }
+
+#if 0
```
Comment (hyperlogic) [inline libraries/baking/src/FBXBaker.cpp]: Delete this `#if 0` code.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9933/HighFidelity-Beta-PR13127-7f2a90cdb9395b399eb3c4409d5e067add1e5860.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9933/HighFidelity-Beta-PR13127-7f2a90cdb9395b399eb3c4409d5e067add1e5860.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9933/HighFidelity-Beta-Interface-PR13127-7f2a90cdb9395b399eb3c4409d5e067add1e5860.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9933/HighFidelity-Beta-PR13127-7f2a90cdb9395b399eb3c4409d5e067add1e5860.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9933/HighFidelity-Beta-Interface-PR13127-7f2a90cdb9395b399eb3c4409d5e067add1e5860.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Fail @conklin94122 
On the GTX 1070 w/ i7-7700 machine it crashes on launch. On the Mac and all AMD machine it fails to load almost all objects.
[CRASHhifi-log.txt](https://github.com/highfidelity/hifi/files/2018726/CRASHhifi-log.txt)
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2018727/hifi-log.txt)
asset_browser_failure
capture

**Comment by samcake** [comment]
pushing milestone to 69

**Comment by SimonWalton-HiFi** [comment]
I'm not sure why this failed; my local build works ok and the May 8 WIndows installer works, but the latest installer has a lot of problems.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9946/HighFidelity-Beta-PR13127-7f2a90cdb9395b399eb3c4409d5e067add1e5860.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9946/HighFidelity-Beta-PR13127-7f2a90cdb9395b399eb3c4409d5e067add1e5860.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9946/HighFidelity-Beta-Interface-PR13127-7f2a90cdb9395b399eb3c4409d5e067add1e5860.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9946/HighFidelity-Beta-PR13127-7f2a90cdb9395b399eb3c4409d5e067add1e5860.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9946/HighFidelity-Beta-Interface-PR13127-7f2a90cdb9395b399eb3c4409d5e067add1e5860.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9951/HighFidelity-Beta-PR13127-e4130dbf39139b4c88fa6193c162d4500bff42cd.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9951/HighFidelity-Beta-PR13127-e4130dbf39139b4c88fa6193c162d4500bff42cd.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9951/HighFidelity-Beta-Interface-PR13127-e4130dbf39139b4c88fa6193c162d4500bff42cd.exe

**Comment by SimonWalton-HiFi** [comment]
The FBX file header change was incorrect - the full prolog is checked for reading files, but existing baked files won't have the new subsection. I've split the prolog in two parts, writing both but checking just the first upon read. May need dev approval again.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9969/HighFidelity-Beta-PR13127-c205a856a71376b91a555b59d3a4c1f39db3de9f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9969/HighFidelity-Beta-PR13127-c205a856a71376b91a555b59d3a4c1f39db3de9f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9969/HighFidelity-Beta-Interface-PR13127-c205a856a71376b91a555b59d3a4c1f39db3de9f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9969/HighFidelity-Beta-PR13127-c205a856a71376b91a555b59d3a4c1f39db3de9f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9969/HighFidelity-Beta-Interface-PR13127-c205a856a71376b91a555b59d3a4c1f39db3de9f.exe

**Comment by qlipdahlin** [comment]
Currently Testing

**Comment by qlipdahlin** [comment]
@conklin94122 PASS

Tested on Desktop - The Mason_Idle.fbx file baked and loaded without issue. Restarts and other steps to cause baked item to crash the interface did not occur. Issue with previous build crashing the GTX machine has been resolved as well.

**Pull request merged**